---
title: Advertencia del compilador (nivel 1) C4930
ms.date: 11/04/2016
f1_keywords:
- C4930
helpviewer_keywords:
- C4930
ms.assetid: 89a206c9-c536-4186-8e81-1cde3e7f4f5b
ms.openlocfilehash: 15cd1ed61c747e2c9168b9fc0fee03dca8403a24
ms.sourcegitcommit: 0ab61bc3d2b6cfbd52a16c6ab2b97a8ea1864f12
ms.translationtype: MT
ms.contentlocale: es-ES
ms.lasthandoff: 04/23/2019
ms.locfileid: "62242791"
---
# <a name="compiler-warning-level-1-c4930"></a>Advertencia del compilador (nivel 1) C4930

'prototipo': no se llama a la función prototipo (era pensada una definición de variable?)

El compilador detectó un prototipo de función no utilizada. Si el prototipo debía entenderse como una declaración de variable, quite los paréntesis de apertura y cierre.

El ejemplo siguiente genera el error C4930:

```
// C4930.cpp
// compile with: /W1
class Lock {
public:
   int i;
};

void f() {
   Lock theLock();   // C4930
   // try the following line instead
   // Lock theLock;
}

int main() {
}
```

C4930 también puede producirse cuando el compilador no puede distinguir entre una declaración de prototipo de función y una llamada de función.

El ejemplo siguiente genera el error C4930:

```
// C4930b.cpp
// compile with: /EHsc /W1

class BooleanException
{
   bool _result;

public:
   BooleanException(bool result)
      : _result(result)
   {
   }

   bool GetResult() const
   {
      return _result;
   }
};

template<class T = BooleanException>
class IfFailedThrow
{
public:
   IfFailedThrow(bool result)
   {
      if (!result)
      {
         throw T(result);
      }
   }
};

class MyClass
{
public:
   bool MyFunc()
   {
      try
      {
         IfFailedThrow<>(MyMethod()); // C4930

         // try one of the following lines instead
         // IfFailedThrow<> ift(MyMethod());
         // IfFailedThrow<>(this->MyMethod());
         // IfFailedThrow<>((*this).MyMethod());

         return true;
      }
      catch (BooleanException e)
      {
         return e.GetResult();
      }
   }

private:
   bool MyMethod()
   {
      return true;
   }
};

int main()
{
   MyClass myClass;
   myClass.MyFunc();
}
```

En el ejemplo anterior, el resultado de un método que toma ningún argumento se pasa como argumento al constructor de una variable de clase local sin nombre. La llamada se puede eliminar la ambigüedad por la variable local de nombres o un prefijo de la llamada al método con una instancia de objeto junto con el operador de puntero a miembro adecuado.